### PR TITLE
Convert to ES6 compatible library

### LINKED ISSUE #15: Moving from ES5 to ES6
Since almost all major JavaScript bundlers have better built-in code splitting, tree-shaking and compile time optimizations in place. it is ideal to have an ES6 version of the package along with the current ES5 version.

### PR DESCRIPTION
**Pull request by AStoker**
# ES6 Conversion + Features

Resolves #15 
Added a new file (`toastify-es.js`) which contains an ES6 compliant version of Toastify.  
Few improvements and changes to behavior:  
- Based off comments, `positionLeft` was finally removed
- Expanded functionality for selector that allows you to pass in an element as well (defaults to `document.body`)

Notable changes to code:
- Made use of arrow functions to keep context the same (replaced `(function(){...}).bind(this)`)
- `let` instead of `var`
- Toastify class rather than function. Exported function creates an instance of Toastify
- When checking classes, use the built in `classList` functionality on elements 
- `rootElement` is now stored on the class so when we remove toasts we know where to look specifically (this enhanced the ability of putting toasts inside elements and allows you to use Toastify inside shadowDom nodes)
- Added ability to use `escapeMarkup` to allow optionally unsafe `innerHTML`
- Added ability to use the HTML DOM `style` property to directly style the toast (this further extends `backgroundColor` functionality, and therefore `backgroundColor` should be deprecated in the future in favor of using `style.background`). Note, since previous functionality took `backgroundColor` and applied it to the elements `background` style (which changes all background styles), to prevent breaking changes, `backgroundColor` takes precedence over `style.backgroundColor`.

Things that should still happen:
- [x] Get maintainer feedback (I'm only human and spent a few hours quickly changing things, don't want to miss something)
- [x] Prevent XSS issues when we pass `text` into the `innerHTML` of the toast (probably just want to use `innerText`, as there's already `node` for when you want to pass in an element)
- [x] Cleanup of default values and states
- [x] Adequate testing of all use cases
- [x] Test across browsers
- [x] Include added functionality in legacy versions

### CODE REVIEW

File: src/toastify-es.js [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+     */
+    _buildToast() {
+      // Validating if the options are defined
+      if (!this.options) {
+        throw "Toastify is not initialized";
+      }
+  
+      // Creating the DOM object
+      let divElement = document.createElement("div");
+      divElement.className = "toastify on " + this.options.className;
```
Comment (apvarun) [inline src/toastify-es.js:183]: Prefer template strings?
Maybe try [Lelab](https://github.com/lebab/lebab) for some more quick changes :)

File: src/toastify-es.js [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+    }
+  
+  }
+  
+  /**
+   * Helper function to get offset
+   * @param {string} axis - 'x' or 'y'
+   * @param {ToastifyConfigurationObject} options - The options object containing the offset object
+   */
+  function getAxisOffsetAValue(axis, options) {
```
Comment (apvarun) [inline src/toastify-es.js:418]: This could just be a private function of the class

**Review by apvarun — COMMENTED** [review]
Awesome work @AStoker 
This is a nice first step. We can later convert this to TS and also generate types file 👍

File: src/toastify-es.js (hunk 1 above)
Comment (AStoker) [reply src/toastify-es.js:183]: I'm all for template strings! I'll take a gander at Lelab too, haven't used it before.

**Comment by AStoker** [comment]
@apvarun, mind giving this PR the once over again? I think it's ready to rock.

**Comment by AStoker** [comment]
@apvarun Okay, now I think this should be ready to rock. Mind giving it the look over?

**Comment by AStoker** [comment]
@apvarun I think this is ready for review

**Comment by AStoker** [comment]
@apvarun, see anything I missed in this PR? Would love to get this first draft in. Next steps can be to create a Typescript version which can compile out to native modules or legacy javascript

**Comment by AStoker** [comment]
@apvarun , let me know if there's something I can do to help this get merged in faster

**Review by apvarun — APPROVED** [review]
Code changes looks fine to me.

**Comment by apvarun** [comment]
I think we should also reference this new `es` file somehow so that the correct file is picked up by bundler

File: src/toastify-es.js [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+      let divElement = document.createElement("div");
+      divElement.className = `toastify on ${this.options.className}`;
+  
+      // Positioning toast to left or right or center (default right)
+      divElement.className += ` toastify-${this.options.position}`;
+  
+      // Assigning gravity of element
+      divElement.className += ` ${this.options.gravity}`;
+  
+      if (this.options.backgroundColor) {
```
[the remark this answers is not in the document]
Comment (AStoker) [reply src/toastify-es.js:198]: Yup, I'll move it up. That block originally had some logic in it, but it got refactored and this wasn't moved. I'll pop it up in init

**Review by ferdragonf-spec — COMMENTED** [review]
Hhh

**Pull request merged**